Finder/TranslationFinder.php

```diff
@@ -54,59 +56,52 @@ public function all()
         $finder = new Finder();
         $finder
             ->files()
-            ->in($this->getLocations())
             ->followLinks();
 
+        $finder = $this->getFinderByFilenameAndFolder($finder);
+
         return $finder;
     }
 
     /**
-     * Gets translation files location.
+     * Get finder with all the filenames and paths.
      *
-     * @return array
+     * @param Finder $finder finder
+     *
+     * @return $this|Finder|\Symfony\Component\Finder\SplFileInfo[] finder instance
      */
-    protected function getLocations()
+    private function getFinderByFilenameAndFolder(Finder $finder)
     {
-        $locations = array();
-
-        if (class_exists('Symfony\Component\Validator\Validation')) {
-            $r = new \ReflectionClass('Symfony\Component\Validator\Validation');
-
-            $locations[] = dirname($r->getFilename()).'/Resources/translations';
-        }
-
-        if (class_exists('Symfony\Component\Form\Form')) {
-            $r = new \ReflectionClass('Symfony\Component\Form\Form');
+        $dirToSearch = [];
 
-            $locations[] = dirname($r->getFilename()).'/Resources/translations';
+        foreach ($this->translationFilesByLocale as $locale => $resourceFilePaths) {
+            foreach ($resourceFilePaths as $filename) {
+                $dirToSearch[] = dirname($filename);
+                $finder = $finder
+                    ->name(basename($filename));
+            }
         }
 
-        if (class_exists('Symfony\Component\Security\Core\Exception\AuthenticationException')) {
-            $r = new \ReflectionClass('Symfony\Component\Security\Core\Exception\AuthenticationException');
+        $finder
+            ->in(array_unique($dirToSearch));
 
-            if (file_exists($dir = dirname($r->getFilename()).'/../../Resources/translations')) {
-                $locations[] = $dir;
-            } else {
-                // Symfony 2.4 and above
-                $locations[] = dirname($r->getFilename()).'/../Resources/translations';
-            }
-        }
+        return $finder;
+    }
 
-        $overridePath = $this->kernel->getRootDir() . '/Resources/%s/translations';
-        foreach ($this->kernel->getBundles() as $bundle => $class) {
-            $reflection = new \ReflectionClass($class);
-            if (is_dir($dir = dirname($reflection->getFilename()).'/Resources/translations')) {
-                $locations[] = $dir;
-            }
-            if (is_dir($dir = sprintf($overridePath, $bundle))) {
-                $locations[] = $dir;
+    /**
+     * Get all folders where there are translation files.
+     *
+     * @return array folders with translations
+     */
+    private function getFoldersWithTranslationFiles()
+    {
+        $dirToSearch = [];
+        foreach ($this->translationFilesByLocale as $locale => $resourceFilePaths) {
```

Comment:
The translationFilesByLocale comes from the FrameworkBundle and it is organized by locale => list of files in an array in the specific locale 